bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoDimmerHandler.java

```diff
@@ -119,13 +116,46 @@ public void initialize() {
     public void dispose() {
         logger.debug("WeMoDimmerHandler disposed.");
 
-        ScheduledFuture<?> job = refreshJob;
+        ScheduledFuture<?> job = this.pollingJob;
         if (job != null && !job.isCancelled()) {
             job.cancel(true);
         }
-        refreshJob = null;
+        this.pollingJob = null;
 
         removeSubscription();
+        service.unregisterParticipant(this);
+    }
+
+    private void poll() {
+        synchronized (jobLock) {
+            if (pollingJob == null) {
+                return;
+            }
+            try {
+                logger.debug("Polling job");
+
+                // First check if the Wemo device is set in the UPnP service registry
+                // If not, set the thing state to ONLINE/CONFIG-PENDING and wait for the next poll
+                if (!isUpnpDeviceRegistered()) {
+                    logger.debug("UPnP device {} not yet registered", getUDN());
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "@text/config-status.pending.device-not-registered [\"" + getUDN() + "\"]");
+                    synchronized (upnpLock) {
+                        subscriptionState = new HashMap<>();
+                    }
+                    return;
+                }
+                if (host.isEmpty()) {
+                    URL descriptorURL = service.getDescriptorURL(this);
+                    host = getWemoHost(descriptorURL);
```

Comment:
First, `NullPointerException` here if **ipAddress** is not configured - see other thread. But also:
- As I understand from you, `service.getDescriptorURL()` can return null?
- In this case null is passed to `getWemoHost()`, which in turn will return empty string (I don't like this, but that's a separate issue).
- So host will still be empty. This can happen in the scenario this PR is fixing, but when user still hasn't configured an IP address.
- Now `updateWemoState()` will silently return as it also checks if host is empty.

I think this should be considered some kind of error and logged?

Reply:
Thanks for investigating.
It might be better to remove getWemoHost() from WemoUtil class again and read the logic to the handler.
I am also thinking of making IP address a mandatory config param. We then could remove some more checks.
We need to highlight this as a breaking change for textual config, but that should not be a big problem…

Reply:
> It might be better to remove getWemoHost() from WemoUtil class again and readd the logic to the handler.

I agree. My other comment probably drowned as it was towards a resolved comment. But after removing the custom parsing and simply using `URL.getHost()` there's not much point in having it in the utility class anymore.

> I am also thinking of making IP address a mandatory config param. We then could remove some more checks.
> We need to highlight this as a breaking change for textual config, but that should not be a big problem…

This is a breaking change that can be avoided, so I would probably opt for keeping backwards compatibility. Instead maybe let's try to consolidate this logic (with corresponding checks) instead of having to do them in each handler? I'm still not sure exactly what is the scenario where IP address **needs** to be manually configured.

When can descriptor URL be obtained from the service? Right after `registerParticipant()`? Maybe a method like this could be reused for all handlers:

```
    public String initializeHost(Configuration configuration, UpnpIOService service) {
        String host = (String) configuration.get(IPADDRESS);
        if (host != null) {
            return host;
        }
        host = service.getDescriptorURL(this).getHost();
        if (host != null) {
            return host;
        }
        return null;
    }

```
and then called from initialize at appropriate place:

```
            host = initializeHost(configuration, service);
            if (host == null) {
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                "Unable to obtain host, please manually configure");
                return;
            }
```

In this case thing will be offline and no jobs will be scheduled. So we should be able to assume that host will be available in methods where it's needed. Let me know what you think, this is just some quick thoughts and I might have missed important details.

Reply:
I think I have found a better/easier solution and can read the descriptorURL in WemoHandlerFactory and pass the host directly into the handler.
Will provide a new commit with according changes tomorrow…